library/psa_crypto_ecp.c

```diff
@@ -32,6 +32,66 @@
     defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH)
+/* Helper function to verify if the provided EC's family and key bit size are valid.
+ *
+ * Note: "bits" parameter is used both as input and output and it might be updated
+ *       in case provided input value is not multiple of 8 ("sloppy" bits).
+ */
+static int check_ecc_parameters(psa_ecc_family_t family, size_t *bits)
+{
+    switch (family) {
+        case PSA_ECC_FAMILY_SECP_R1:
+            switch (*bits) {
+                case 192:
+                case 224:
+                case 256:
+                case 384:
+                case 521:
+                    return PSA_SUCCESS;
+                case 528:
+                    *bits = 521;
+                    return PSA_SUCCESS;
+            }
+            break;
+
+        case PSA_ECC_FAMILY_BRAINPOOL_P_R1:
+            switch (*bits) {
+                case 256:
+                case 384:
+                case 512:
+                    return PSA_SUCCESS;
+            }
+            break;
+
+        case PSA_ECC_FAMILY_MONTGOMERY:
+            switch (*bits) {
+                case 448:
+                case 255:
+                    return PSA_SUCCESS;
+                case 256:
+                    *bits = 255;
+                    return PSA_SUCCESS;
+            }
+            break;
+
+        case PSA_ECC_FAMILY_SECP_K1:
+            switch (*bits) {
+                case 192:
+                case 224:
+                case 256:
+                    return PSA_SUCCESS;
+                /* secp224k1 has 224-bit coordinates but 225-bit private keys.
+                 * This means that private keys are represented with 232 bits. */
+                case 232:
```

Comment:
Our PSA implementation [does not and will not support secp224k1](https://github.com/Mbed-TLS/mbedtls/issues/3541). So we could save a few bytes of code size and maybe a little bit of complexity by removing the existing support. (Keep a comment, since it would be somewhat surprising that secp224k1 is missing in the PSA code even though it's supported in the legacy API.)

For `check_ecc_parameters`, we just have to return `INVALID_ARGUMENT` or `NOT_SUPPORTED`. For `mbedtls_ecc_group_from_psa` and `mbedtls_ecc_group_to_psa`, we need to make sure the behavior is consistent with what we say about recognized curves in the documentation.

Reply:
I suspect that removing the support for secp224k1 might end up in the failure I mentioned in a [previous comment](https://github.com/Mbed-TLS/mbedtls/pull/8664#issuecomment-1871849371), in particular:
> those tests are supposed to be skipped (according to their description), but they are executed because SECP224K1 is commented out in crypto_config.h. Perhaps also this thing might be fixed sooner or later in this automatic generated test.